content/en/docs/ibm/deploy/authentication.md

```diff
@@ -145,3 +145,130 @@ The secret in the `default` namespace is automatically renewed by IBM Cloud
+    Be sure to update `<hostname>`:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      <hostname>: ""
+      _example: |
+       ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Then, save and exit. The routes for your InferenceServices will start using this new domain.
+
+2. Since the certificates provided by IBM Cloud only allow for a single level of domain name added
+to the base domain, the domain template for Knative needs to be adjusted so that the certificates
+will be valid for the InferenceService routes.
+
+    ```shell
+    kubectl edit configmap config-network -n knative-serving
+    ```
+
+    This will open your default text editor, and you will again see something like:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      _example: |
+        ################################
+        #                              #
+        #    EXAMPLE CONFIGURATION     #
+        #                              #
+        ################################
+        # ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Add a line above the `_example` key with the `domainTemplate` key like the following:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      domainTemplate: "{{.Name}}-{{.Namespace}}.{{.Domain}}"
+      _example: |
+       ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Save and exit. The default template uses a two-level subdomain (i.e. `{{.Name}}.{{.Namespace}}.{{.Domain}}`),
+    so this just adjusts it to use one.
+
+
+3. Adjust `kubeflow-gateway` one more time, adding a wildcard host in the HTTPS `hosts` section.
+
+    ```shell
+    kubectl edit gateway kubeflow-gateway -n kubeflow
+    ```
+
+    This will open your default text editor, but you can also optionally edit `kubeflow-gateway.yaml` file
+    you created previously. Here, just add another entry to the HTTPS `hosts` list containing your hostname prepended
+    with a `*.` so that the Knative subdomains are properly routed.
+
+    ```YAML
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: kubeflow-gateway
+      namespace: kubeflow
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+      - hosts:
+        - '<hostname>'
+        - '*.<hostname>'
+        port:
+          name: https
+          number: 443
+          protocol: HTTPS
+        tls:
+          mode: SIMPLE
+          privateKey: /etc/istio/ingressgateway-certs/tls.key
+          serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+    ```
+
+    Save and exit.
+
+After these adjustments, InferenceServices should now be reachable via HTTPS. However, if you are using IBM Cloud AppID for
+authentication and you want to predict externally, you will need to generate a token as
+[described here](https://cloud.ibm.com/docs/appid?topic=appid-custom-auth#generating-jwts). This token can be used as a header
+(e.g. `"Authorization: Bearer eyJh..."`) in your request.
```

Comment:
Do you mind adding an example about how to use this token to access kfserving? I'm curious which component is validating this JWT token from AppID since the envoyfilter won't allow `Authorization` header go through.

Reply:
Ah sorry, good catch. I didn't realize that the environment I was using was further customized with the envoyFilter including the "Authorization" pattern in the allowed_headers, and a newer authservice image (http://gcr.io/arrikto/oidc-authservice:087a340) was being used (one that includes a recent PR https://github.com/arrikto/oidc-authservice/commit/087a340cb454766891fdd3092955954c06d154ac).

In that case, I think I may just leave this JWT section out for now.

Reply:
@pvaneck that's what I guessed. Yes, it will work with the latest oidc-authservice which depends on https://github.com/kubeflow/kubeflow/issues/5302